hub/core/transform/transform_tensor.py

```diff
@@ -34,12 +36,26 @@ def __len__(self) -> int:
             self.numpy_compressed()  # calculates and sets length
         return self.length  # type: ignore
 
+    def __getattr__(self, name):
+        if self.items:
+            raise AttributeError(name)
+        del self.dataset.tensors[self.name]
```

Comment:
could you leave a comment in the code explaining that this method? it's not very obvious what's going on.